src/pages/filter/ui/panels/function-panel/components/compound-het/compound-het.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import { Option } from 'react-dropdown'
+import { observer } from 'mobx-react-lite'
+
+import { IStatFuncData } from '@declarations'
+import { FuncStepTypesEnum } from '@core/enum/func-step-types-enum'
+import filterStore from '@store/filter'
+import { DropDown } from '@ui/dropdown'
+import { DisabledVariantsAmount } from '@pages/filter/ui/query-builder/ui/disabled-variants-amount'
+import functionPanelStore from '../../function-panel.store'
+import { PanelButtons } from '../panelButtons'
+import compoundHetStore, {
+  CompoundHetSelectOprions,
+} from './compound-het.store'
+
+export const CompundHet = observer((): ReactElement => {
+  const { cachedValues, initialApprox } = compoundHetStore
+
+  const { simpleVariants, filterName, filterGroup } = functionPanelStore
+
+  const [statFuncStatus, setStatFuncStatus] = useState<IStatFuncData>()
+
+  const fetchStatFuncAsync = async () => {
```

Comment:
I think there is no reason to create `fetchStatFuncAsync` in component, you can inline it in useEffect callback: `functionPanelStore.fetchStatFunc(...).then(setStatFuncStatus, ...)`.
Also there is no error handling now.
And it is a bad practice to use `setState` in a async response callback. Because it can be called when component is already unmounted.

May be we can use store for this?